tests/test_models.py

```diff
@@ -373,6 +373,30 @@ def test_timdex_record_not_a_list_raises_error(timdex_record_required_fields):
         timdex_record_required_fields.dates = "test"
 
 
+def test_timdex_object_hash_diff_if_diff_class():
+    """
+    Asserts that TIMDEX objects of different class types
+    with similar attributes and attribute values will
+    be assigned different hashes and declared not equal.
+    """
+    identifier = timdex.Identifier(value="x", kind="y")
+    alternate_title = timdex.AlternateTitle(value="x", kind="y")
+    assert identifier != alternate_title
+    assert identifier.__hash__() != alternate_title.__hash__()
```

Comment:
Nice test!  I like how explicit these are to test this pretty important little kernel of code and logic.  

Yes, perhaps we _implicitly_ test this in other ways, but it's nice to have this.

Reply:
Agree these are very good tests, great work!